app/models/search.rb

```diff
@@ -146,13 +172,19 @@ def search
       self.results[type] = _results.results
       self.hits[type]    = _results.hits
       self.total_entries[type] = _results.results.total_entries
+      _results.facet(:subject_areas).rows.each do |facet|
+        self.availabe_subject_areas << facet.value
+      end
+      _results.facet(:grade_level_groups).rows.each do |facet|
+        self.availabe_grade_level_groups[facet.value] = 1
+      end
```

Comment:
Because you are running these facets on the the results of the search, then the `available_grade_level_groups` and `available_subject_areas` will be restricted to what materials match the current search criteria.
Instead of this, the `available_subject_areas` and `available_grade_level_groups` should be based on all of the materials in the database.  So you should do a separate search that matches everything, and add these facets to that search.